benchmark/SoftMaxCrossEntropyBenchmark.jl

```diff
@@ -0,0 +1,70 @@
+include("../src/layers/LayerBase.jl")
+include("../src/layers/SoftMaxCrossEntropy.jl")
+include("../src/layers/SoftMax.jl")
+include("../src/layers/CrossEntropyLoss.jl")
+using Base.Test
+
+smaxcross = SoftMaxCrossEntropyLoss()
+smax = SoftMax()
+cross = CrossEntropyLoss()
+
+
+function before(smaxcross, smax, cross, dict)
+    init(smaxcross, nothing, dict)
+    init(smax     , nothing, dict)
+    init(cross    , nothing, dict)
+end
+
+function convert_to_one_hot(x::Array{Int64}, classes)
+  m = zeros(size(x,1), classes)
+  for i=1:size(x,1)
+    m[i,x[i]+1]=1
+  end
+  m
+end
+
+function benchmmark(smaxcross, smax, cross, batch_size, input_size; alpha = 1.)
+  # batch_size = 10
+  # input_size = 5
+  before(smaxcross, smax, cross, Dict{String, Any}("batch_size" => batch_size, "input_size" => [input_size]))
+  x = rand(batch_size, input_size)
+  label = rand(0:input_size-1, batch_size,1)
+  onehot_label = convert_to_one_hot(label,input_size)
+  # benchmark for original SoftMaxCrossEntropy
+  tic()
```

Comment:
I still think that as a benchmark you should be able to give both the execution time and memory allocation, since both will be metrics to be benchmarked. In this case, `@times` will be a better option, since it could be used to accomplish both.